File: apps/python/routes/body/blob.py
from primate import Route


@Route.post(content_type="application/octet-stream")
def handle_post(request):
    blob = request.body.blob()
    return {
        "type": blob.content_type,
        "size": blob.size,
        "head": blob.head(4),
    }

File: apps/ruby/routes/body/blob.rb
require 'primate/route'

Route.post(content_type: "application/octet-stream") do |request|
  blob = request.body.blob
  {
    "type" => blob.content_type,
    "size" => blob.size,
    "head" => blob.head(4),
  }
end
